packages/yarnpkg-pnp/sources/generatePnpScript.ts

```diff
@@ -44,7 +44,8 @@ function generateInlinedSetup(data: SerializedState) {
 function generateSplitSetup() {
   return [
     `function $$SETUP_STATE(hydrateRuntimeState, basePath) {\n`,
-    `  return hydrateRuntimeState(require(${JSON.stringify(`./${Filename.pnpData}`)}), {basePath: basePath || __dirname});\n`,
+    `  const fs = require('fs');\n`,
+    `  return hydrateRuntimeState(JSON.parse(fs.readFileSync(${JSON.stringify(`./${Filename.pnpData}`)}, 'utf8')), {basePath: basePath || __dirname});\n`,
```

Comment:
`fs.readFileSync` is relative to the current working directory so you need to use `path.join` and `__dirname` here.

Reply:
Ugh, good catch! That's exactly what it used to look like, and I probably lost it during rebase: https://github.com/yarnpkg/berry/compare/%40yarnpkg/cli/3.6.1...shYkiSto:berry:3.6.1-patch-pnpapi?expand=1#diff-7e72010f850cd7d793fa4c212795befc8c848dc226efbee46c04940c5230e5be

Fixed..